eng/publish/publish.yml

```diff
@@ -0,0 +1,281 @@
+      # DTFx.Core upload task would fail (as expected, it was already uploaded), 
+      # but the DTFx.AzureStorage upload would succeed.
+
+      # Implementor's note:
+      # What follows could use some refactoring as there's a lot of repetition, perhaps an ADO template could hide the repeated details.
+      # Pay close attention to the `packagesToPush` property of each job. We need to make sure the pattern specified there uniquely identifies
+      # the package we want to push, and this is difficult when the full name of one package is the prefix of another.
+      # For example: `Microsoft.DurableTask.Client` and `Microsoft.DurableTask.Client.Grpc`.
+      # The trick is to exclude the `Microsoft.DurableTask.Client.Grpc` in the `packagesToPush` property for `Microsoft.DurableTask.Client`
+
+      # NuGet release (Microsoft.DurableTask.Abstractions)
+      - job: nugetRelease_Microsoft_DurableTask_Abstractions
+        displayName: NuGet Release (Microsoft.DurableTask.Abstractions)
+        dependsOn: nugetApproval
+        condition: succeeded('nugetApproval') # nuget packages need to be on ADO first
+        templateContext:
+          type: releaseJob
+          isProduction: true
+          inputs:
+          - input: pipelineArtifact
+            pipeline: officialPipeline  # Pipeline reference as defined in the resources section
+            artifactName: drop
+            targetPath: $(System.DefaultWorkingDirectory)/drop
+        steps:
+        - task: 1ES.PublishNuget@1
+          displayName: 'NuGet push (Microsoft.DurableTask.Abstractions)'
+          inputs:
+            command: push
+            nuGetFeedType: external
+            publishFeedCredentials: 'DurableTask org NuGet API Key'
+            packagesToPush: '$(System.DefaultWorkingDirectory)/drop/Microsoft.DurableTask.Abstractions.*.nupkg;!$(System.DefaultWorkingDirectory)/**/*.symbols.nupkg' # Despite this being a custom command, we need to keep this for 1ES validation
+            packageParentPath: $(System.DefaultWorkingDirectory) # This needs to be set to some prefix of the `packagesToPush` parameter. Apparently it helps with SDL tooling
+
+      # NuGet release (Microsoft.DurableTask.Client)
+      - job: nugetRelease_Microsoft_DurableTask_Client
+        displayName: NuGet Release (Microsoft.DurableTask.Client)
+        dependsOn: nugetApproval
+        condition: succeeded('nugetApproval') # nuget packages need to be on ADO first
+        templateContext:
+          type: releaseJob
+          isProduction: true
+          inputs:
+          - input: pipelineArtifact
+            pipeline: officialPipeline  # Pipeline reference as defined in the resources section
+            artifactName: drop
+            targetPath: $(System.DefaultWorkingDirectory)/drop
+        steps:
+        - task: 1ES.PublishNuget@1
+          displayName: 'NuGet push (Microsoft.DurableTask.Client)'

```

Comment:
Splitting these all into separate jobs looks quite cumbersome, especially now we will need to edit this every time a new project is added.

In legacy release, I worked around this by using the dotnet CLI for pushing to nuget.org. Did you try using `useDotNetTask: true` and `allowPackageConflicts: true`?

Hopefully with those 2 inputs this can be consolidated down to a single job.

Reply:
Oh, I don't recall seeing a `useDotNetTask` option, perhaps that's what I need. I can give it a go.
The key issue is that the 1ES  nuget release task has a limitation of not allowing package conflicts, so if this setting allows me to go back to the original nuget release task, then this would be it